### PR TITLE
Add alpha formatter support

### PR DESCRIPTION
**Pull request by zanieb**
Tested with `pre-commit try-repo . --all-files`

```
❯ pre-commit try-repo . --all-files
[INFO] Initializing environment for ..
===============================================================================
Using config:
===============================================================================
repos:
-   repo: .
    rev: 6bd1079860e4a235eb49e2d56857d43748f7a389
    hooks:
    -   id: ruff
    -   id: ruff-format
===============================================================================
[INFO] Installing environment for ..
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
ruff.....................................................................Passed
ruff-format..............................................................Failed
- hook id: ruff-format
- exit code: 1

warning: `ruff format` is a work-in-progress, subject to change at any time, and intended only for experimentation.
1 file would be reformatted, 1 file left unchanged

```

And a configuration pointing to this work

```yaml
repos:
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: 2eeece5511efd959268030fa80ece7a2d54411b5
    hooks:
      - id: ruff
      - id: "ruff-format"
        args: [--check]
```
```
❯ pre-commit run --all-files
[INFO] Installing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
ruff.....................................................................Passed
ruff-format..............................................................Passed
```

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+  language: python
+  "types_or": [python, pyi]
+  args: []
+  require_serial: true
+  additional_dependencies: []
+  minimum_pre_commit_version: "2.9.2"
+- id: ruff-format
+  name: ruff-format
+  description: "Run 'ruff format' for extremely fast Python formatting"
+  entry: ruff format --check
```
Comment (charliermarsh) [inline .pre-commit-hooks.yaml:14]: Probably want force-exclude here too

Comment (zanieb) [reply .pre-commit-hooks.yaml:14]: Ah I wasn't sure it was supported. Great!

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+  language: python
+  "types_or": [python, pyi]
+  args: []
+  require_serial: true
+  additional_dependencies: []
+  minimum_pre_commit_version: "2.9.2"
+- id: ruff-format
+  name: ruff-format
+  description: "Run 'ruff format' for extremely fast Python formatting"
+  entry: ruff format --check --force-exclude
```
Comment (zanieb) [inline .pre-commit-hooks.yaml:14]: Doesn't format files but probably should by default?

File: README.md [hunk 3]
```diff
@@ -50,6 +50,28 @@ _unless_ you enable autofix, in which case, Ruff's pre-commit hook should run _b
 and other formatting tools, as Ruff's autofix behavior can output code changes that require
 reformatting.
 
+### Using Ruff's formatter (unstable)
+
+Ruff's formatter is in Alpha, but can used with pre-commit by adding the `ruff-format` hook:
```
Comment (charliermarsh) [inline README.md:55]: Nit: link to the README?

File: .pre-commit-hooks.yaml [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
+- id: ruff-format
+  name: ruff-format
+  description: "Run 'ruff format' for extremely fast Python formatting"
+  entry: ruff format --force-exclude
+  language: python
+  "types_or": [python, pyi]
+  args: []
+  require_serial: true
+  additional_dependencies: []
+  minimum_pre_commit_version: "2.9.2"
```
Comment (charliermarsh) [inline .pre-commit-hooks.yaml:20]: What was this formatted with?

**Review by charliermarsh — APPROVED** [review]

File: .pre-commit-hooks.yaml (hunk 4 above)
Comment (zanieb) [reply .pre-commit-hooks.yaml:20]: Good question I didn't even notice it formatted — I think the RedHat YAML Extension in VSCode. Happy to revert or use something else.

File: README.md [hunk 5]
```diff
@@ -50,6 +50,28 @@ _unless_ you enable autofix, in which case, Ruff's pre-commit hook should run _b
...
+[Ruff's formatter](https://github.com/astral-sh/ruff/blob/main/crates/ruff_python_formatter/README.md) is in Alpha, but can used with pre-commit by adding the `ruff-format` hook:
+
+```yaml
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.0.289
+  hooks:
+    - id: ruff-format
+```
+
+To check formatting with changing files,  use `--check`:
```
Comment (mkniewallner) [inline README.md:64]:
```suggestion
To check formatting without changing files, use `--check`:
```

Comment (zanieb) [reply README.md]: Thanks!

File: .pre-commit-hooks.yaml (hunk 4 above)
Comment (zanieb) [reply .pre-commit-hooks.yaml:20]: The other formatting was weird actually just going to merge with this — we can adjust later if desired.

**Pull request merged**